src/saving_tools.jl

```diff
@@ -19,24 +19,35 @@ julia> current_commit()
 
 julia> current_commit(path_to_dirty_repo)
 "3bf684c6a115e3dce484b7f200b66d3ced8b0832_dirty"
+
+julia> current_commit(another_path_with_annotated_git_tags)
+"v1.2.3-g7364ab"
 ```
 """
 function current_commit(gitpath = projectdir())
     # Here we test if the gitpath is a git repository.
     try
         repo = LibGit2.GitRepo(gitpath)
     catch er
-        @warn "The current project directory is not a Git repository, "*
-        "returning `nothing` instead of the commit id."
+        @warn "The directory ('$gitpath') is not a Git repository, "
+              "returning `nothing` instead of the commit ID."
         return nothing
     end
-    # then we return the current commit
-    repo = LibGit2.GitRepo(gitpath)
-    c = string(LibGit2.head_oid(repo))
+    suffix = ""
     if LibGit2.isdirty(repo)
-        @warn "The Git repository is dirty! Adding appropriate comment to "*
-        "commit id..."
-        return c*"_dirty"
+        suffix = "_dirty"
+        @warn "The Git repository is dirty! Appending $(suffix) to the commit ID"
+    end
+    # then we return the output of `git describe` or the latest commit hash
+    # if no annotated tags are available
+    repo = LibGit2.GitRepo(gitpath)
+    c = begin try
```

Comment:
what does the `begin` keyword achieve here?

Reply:
Ah, that's a leftover, I'll remove it!